### PR TITLE
imhex: 1.30.1 -> unstable-2023-10-01

### LINKED ISSUE #258674: Update request: imhex 1.30.1→ 1.31.0
- Package name: imhex
- Latest released version: 1.31.0

- Current version on the unstable channel: 1.30.1
- Current version on the stable/release channel: 1.30.1

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@Luis-Hebendanz 

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by cafkafk**
## Description of changes
Closes: #258674
Changelog: https://github.com/WerWolv/ImHex/releases/tag/v1.31.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cafkafk** [comment]
Seems like this relies on capstone >=5 

https://github.com/WerWolv/ImHex/blob/a62ede784018f9d5aaf40587f71a1271429ab50b/lib/libimhex/include/hex/helpers/disassembler.hpp#L30

https://github.com/capstone-engine/capstone/blob/1fc1011d669c893ecd7cb107d3a8e4976ff10e19/include/capstone/capstone.h#L51

**Comment by cafkafk** [comment]
Probably blocked by #252945

**Comment by cafkafk** [comment]
Maybe it's reasonable to go unstable until next release.

**Comment by cafkafk** [comment]
I moved this to `pkgs/by-name` as well, and added myself as a maintainer (if you're okay with that, if not tell me and I can remove it).

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 258964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>imhex</li>
  </ul>
</details>

**Review by Qubasa — APPROVED** [review]

File: pkgs/applications/editors/imhex/default.nix
Comment (bjornfor) [inline pkgs/applications/editors/imhex/default.nix:26]: Self-referencing FIXME? Why is it unstable?

Comment (cafkafk) [reply pkgs/applications/editors/imhex/default.nix]: Seems to be a mistake, is this ok?
```
  # FIXME: unstable, stable needs #252945 (details in #258964)
  # Next version bump should be stabilized
```

Comment (bjornfor) [reply pkgs/applications/editors/imhex/default.nix]: Looks much better 😀

**Review by Qubasa — APPROVED** [review]

**Pull request merged**
